cmd/sonobuoy/app/args.go

```diff
@@ -183,6 +183,30 @@ func AddDeleteAllFlag(flag *bool, flags *pflag.FlagSet) {
 	)
 }
 
+// AddDeleteWaitFlag adds a boolean flag for waiting for the delete process to complete.
+func AddDeleteWaitFlag(flag *bool, flags *pflag.FlagSet) {
```

Comment:
A little unsure if I should just try and combine the flags, maybe even into their own flagset so they get added together automatically.

Also open to debate about how much to allow customized without flag proliferation:
 --wait
 --wait-timeout

Unsure if we should allow these to be customized; defaults seem sufficient for now:
- polling interval (makes sense to poll delete every 2-5s but run should probably be more like 20-30s default since runs take so much longer)
- grace period (when starting a run, we need to give some time within which to allow errors/allow the pods to come up, but how long do we allow the pending state without erroring?)

Reply:
I'm wondering if we should use just `--timeout` for the flag. Are there any other timeouts that would apply to each action?

Reply:
...that would then we'd rename everything from `waittimeout` to just `timeout` in the code for consistency.

Reply:
So there is another timeout which is within the sonobuoy config; it is the timeout for how long the aggregator will wait before forcing plugins to stop. If you sync those two up then you are being racy and will always short some output since there is actions the server takes after that timeout. Plus, keeping them separate allows a little more flexibility in case you want to wait (but not force plugin failures) then check logs or something like that.

Reply:
As of now that timeout is just in the sonobuoy config (passed via the --config flag) but there is an issue up for making that customizable from the flags. #482

Reply:
I see, thanks for clarifying. I'm not in love with the `wait-timeout` piece. Would there be a way to have just `--wait` which would wait and apply the default, then you could use `--wait=60` to specify the timeout? 

If not, it is what it is. =)

Reply:
Yes I can do that with cobra. sgtm.

Reply:
Modified flags so that it is just one value being tracked (an int) and not two (int and bool).

 - I take use of the 0 value (which I default the flags to) meaning that we should not wait.
 - If the flag is provided but not specified a value: `--wait` then the default value is provided (60m for delete, 24h for run)
 - If the flag is specified a value, it is used as expected